app/models/comment.rb

```diff
@@ -31,6 +32,15 @@ def self.attributes
     ]
   end
 
+  def add_user_as_observer
+    proposal = self.proposal
+    user = self.user
+    if !proposal.is_user_tracking? user
+      observer = Observation.new(user: user, proposal: proposal)
+      observer.save
+    end
+  end
```

Comment:
Seems like this logic should belong in proposal? Something like `self.proposal.subscribe(self.user)`

Also, there's a `proposal.add_observer`